exchanges/gateio/gateio_websocket.go

```diff
@@ -557,18 +557,25 @@ func (g *Gateio) processSpotBalances(data []byte) error {
 	if err != nil {
 		return err
 	}
-	accountChanges := make([]account.Change, len(resp.Result))
+	creds, err := g.GetCredentials(context.Background())
+	if err != nil {
+		return err
```

Comment:
```GateIO websocket balances push data error: GateIO REST or Websocket authentication support is not enabled```

In test `TestBalancesPushData` once the context has been populated you change to this:
```go
ctx := account.DeployCredentialsToContext(context.Background(), &account.Credentials{Key: "test", Secret: "test"})
	if err := g.WsHandleSpotData(ctx, []byte(wsBalancesPushDataJSON)); err != nil {
		t.Errorf("%s websocket balances push data error: %v", g.Name, err)
	}
```

Reply:
Sorry, I missed the exchange test. I’ll fix it.

Reply:
All good legend thanks! ❤️